File: lab2/source/mul.cpp
#include "operations.h"
#include "factory.h"

class Mul : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        int res = previous_top * stack.top();
        stack.pop();
        stack.push(res);
    }

};

namespace {
    Operations* createMul() {
        return new Mul;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string("*"), createMul);
}

File: jaba/lab1/src/main/java/org/example/Mul.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class Mul extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s) {
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        int res = previous_top * stack.pop();
        stack.push(res);
    }
}
